maven-model-builder/src/main/java/org/apache/maven/utils/Os.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.utils;
+
+import java.util.Locale;
+import java.util.stream.Stream;
+
+/**
+ * OS support
+ */
+public class Os {
```

Comment:
Would it make sense to skip this class?

Reply:
Well, the point is to get rid of plexus-utils, so we need a replacement for that one too.

Reply:
Yes, we do, but one thing at a time.

Reply:
@michael-o @elharo so I don't really understand what you both want / imply.  Could you be more specific please?

Reply:
Split out all changes involving the Os class into a separate PR. Do not change the Os class or its usages in this PR, so that easier changes aren't gated on more complex work.

Reply:
Os, DAG and ReflectionValueExtractor should be at least three PRs, not one.